### ISSUE TITLE
The future of this project

### ISSUE THREAD

**Issue by mvsmal** [UnoSD/Pulumi.FSharp.Extensions#20]
Hi there,

First of all I would like to thank you so much for this amazing project. We have been using the libraries quite heavily in our projects. I'd say that we strongly depend on them nowadays.

Unfortunately, looking at the commits history and the open issues it is obvious the you don't have much time for maintenance. 
I was wondering what is the future of this project? Are you planning to put effort into it or you are going to abandon it completely?

If my team offers support on maintenance, how should we better do that? Would you have time to review our PRs? 

Or maybe you can add me/us as collaborators so we can continue keeping the project up to date? In this case I would highly appreciate if you find time for an introduction call, just to go over the project structure and code style conventions. We don't want to spoil all the hard work you've done. We don't want to fork the project and create duplicates in the Nuget repository, we'd rather contribute to the original version.

Please let me know your thoughts. I'd be happy to discuss any details.

**Comment by UnoSD** [comment]
Hi,

thank you, I appreciate my project was helpful. Indeed I have had no time to work on any of my hobby projects, hence this was left few versions behind the Pulumi schema. I'd welcome any PR to keep the project up-to-date.

Maintenance should be fairly easy to keep the version in line with Pulumi as it may just need to bump up the version and kick another build, this could also be automated so no human intervention is required unless there are errors (rare). Happy to jump on a call to show you around the project.

Thank you.

**Comment by mvsmal** [comment]
Hi, thanks for the answer. 
As you have noticed, I've submitted a few PRs already and I am planning to submit more.

As for automation of version upgrades, do you think we can do it with GH Actions? Or do you have some other CI/CD tool in mind? I would be happy to contribute to that as well. But I guess you might also need to spend some time on it. I can't fully implement it myself, obviously.

**Comment by UnoSD** [comment]
Hi @mvsmal,

there is already an Azure DevOps pipeline: https://github.com/UnoSD/Pulumi.FSharp.Extensions/blob/master/FAKE.yml

But you will notice that the pipeline is pretty thin as it merely invokes the FAKE script, I wanted to make it as portable as possible and as much F# as possible everywhere; this means that, if you want to create a GH Actions pipeline, it should be a few minutes' work. Happy to merge the PR for GH Actions, however, if you want to add Docker, you just need to create a new pipeline that exposes the PROVIDER env variable "Docker" and that would just work, see here:

https://github.com/UnoSD/Pulumi.FSharp.Extensions/blob/master/build.fsx

```fsharp
let getProvider args =
    match Map.tryFind "PROVIDER" args |>
          Option.bind (function | Argument p -> Some p | _ -> None) |>
          Option.orElse (Environment.environVarOrNone "PROVIDER") |>
          Option.orElse (Environment.environVarOrNone "BUILD_DEFINITIONNAME") with
    | Some "All" -> "*"
    | Some p     -> p
    | _          -> failwith "Missing provider"
```

**Comment by NatElkins** [comment]
Never mind, I think I understand now.  I needed to pass in "All."  I'm getting another error now, but I will continue to work through it.

**Comment by UnoSD** [comment]
> Never mind, I think I understand now. I needed to pass in "All." I'm getting another error now, but I will continue to work through it.

Yes, that is the name of the provider, E.G. Azure, AzureNative, Kubernetes et cetera; "All" generates all of them. Please bear in mind that this issue is unrelated to your question, if you need to discuss further, please open a new issue.

**Comment by cagyirey** [comment]
Hey @UnoSD, I've been maintaining a fork over at cagyirey/Pulumi.FSharp.Extensions. I'd love to contribute to the core project. (especially to have PR reviewers) but it seemed like this repo was dead. Would love to chat more about the possibility of getting some of my work polished and merged. Some of the things I've been working on:

- Support for `ComponentResource`s, introduced to Kubernetes to support `ConfigFile` and `ConfigGroup` resources
- Fixes to typegen logic to allow the aws provider to use `CustomResource`s as input types (a breaking schema change in >6.49 iirc)
- New providers, fixes to the provider name regex
- New problematic type filters
- New `CustomResourceOption` and `ComponentResourceOption` computation expression operations to enable customization of resources created using CE syntax
- Upgraded to the latest versions of FAKE, Paket and .NET
- MiniScaffold-based build system (although I want to switch to YAML changelogs from markdown)
- GitHub Actions-based CI/CD pipeline w/ scheduled package updates

**Comment by UnoSD** [comment]
hi @cagyirey , thank you for working on this fork, I appreciate this project getting some contribution as I still believe it has great value. For personal reasons, I have not been able to work on it myself, but I'd welcome PRs and I'd be happy to integrate changes and push to NuGet to keep it alive.

**Comment by cagyirey** [comment]
My first PR has been open since April :( Fully understand that you've got a lot of life things going on, but a lot of people are blocked right now from receiving package updates. With respect, it may be best to pass the torch to someone who currently has the capacity, or open the project up to maintainers.

**Comment by UnoSD** [comment]
@cagyirey I will try and check out the PR as soon as I have a chance, apologies if I missed it. I am more than happy to open the project to maintainers, if you are happy to onboard.

**Comment by mvsmal** [comment]
@UnoSD Since you are here, I would be happy to get my [PR](https://github.com/UnoSD/Pulumi.FSharp.Extensions/pull/29) reviewed and merged too 🙂

**Comment by UnoSD** [comment]
@mvsmal reviewing it now, apologies

**Comment by NatElkins** [comment]
@UnoSD Would you accept a PR of @cagyirey's fork?  Just playing around locally, I merged in latest main from this repo into theirs.  Minor issues, pretty much just merging in the CustomResourceOptions CE.  The big changes are that it looks like it publishes to a private NuGet feed, but we can sort all that out.  What do you think, would you be open to taking that PR?

**Comment by UnoSD** [comment]
hi @NatElkins, of course. If you would also like to help review, that would be awesome.

@cagyirey can you please submit the PR?
